lib/interpolateName.js

```diff
@@ -52,6 +52,9 @@ function interpolateName(loaderContext, name, options) {
   let basename = 'file';
   let directory = '';
   let folder = '';
+  const query = loaderContext.resourceQuery
+    ? loaderContext.resourceQuery.substr(1)
+    : '';
```

Comment:
I think we can solve this in easy way, like:
```
const queryStringIdx = targetFile.indexOf('?');
let query = ''; 
if (queryStringIdx >= 0) {
  query= targetFile.slice(queryStringIdx);
}  
```

without

Reply:
I'm not quite sure if I understood correctly. As far as I understand, the `resourcePath` and `resourceQuery` are two separate entities. I can't split the path by `?`, but I was trying to remove the leading `?` character. I updated the code to be a bit more verbose. I hope this is what you meant.

Reply:
Logic is simple, if `name` contains `?foo=bar` we interpreted this as `[query]`

Reply:
Could you elaborate on which `name` you mean? I can only get `?foo=bar` from `loaderContext.resourceQuery`. I feel like I'm missing something.

Reply:
`name` - second argument for `interpolateName`